components/sapi/sapi.c

```diff
@@ -0,0 +1,34 @@
+#include "sapi/sapi.h"
+
+#include <string.h>
+
+typedef datadog_php_sapi sapi_t;
+typedef datadog_php_string_view string_view_t;
+
+#define STRING_VIEW_LITERAL(cstr) \
+    { sizeof(cstr) - 1, cstr }
```

Comment:
Was this left over before creating `DATADOG_PHP_STRING_VIEW_LITERAL`?

Reply:
Yes, but it shortens it quite a bit ahaha. Think I should remove it?

Reply:
I'd love to have a shorter macro for this since we'd use it similarly to `ZEND_STRL` in a lot of places. Could we possibly export a shorter version of the macro and use that instead? I can't think of a really solid short name... `DATADOG_PHP_STRL` maybe? 

What are your thoughts on having a "short" version of the `DATADOG_PHP_` namespace for special cases like this one? Something like `DDPHP_`. Then the macro could be `DDPHP_STRL`.

Reply:
If we make a shorter symbol public, like `DDPHP_`, then we may as well just use it everywhere. All of my typedefs and other shortcuts are always in `.c` files so they aren't exposed.

C makes this a pain; every modern language has a solution for this, and C hasn't moved an inch; the name you see is the symbol name you get (except the compiler may or may not put an underscore in front of it).

Maybe we could use a macro to shorten symbols? Something like `DATADOG_PHP_SHORTEN(STRING_VIEW_LITERAL)` which defines a macro in the current file which maps `STRING_VIEW_LITERAL` to `DATADOG_PHP_STRING_VIEW_LITERAL`?

Not sure the macro is better than just writing each one ad-hoc...

Reply:
We decided that doing typedef and defining macros in each file that wants them is what we should do for now. Generally chop off `datadog_php` or `DATADOG_PHP` and append `_t` to types. Not sure how to define rules for things like `SV` -> `DATADOG_PHP_STRING_VIEW_LITERAL`.

Problem goes away if we move it to Rust, ha 🚀 .